src/App.tsx

```diff
@@ -15,81 +19,144 @@ export const goodsFromServer = [
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+  lengthSort = (event: React.MouseEvent<HTMLElement>) => {
+    const target = event.target as Element;
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
-};
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+
+    if (target.classList.contains('is-light')) {
+      this.setState({
+        showResetButton: true,
+      });
+    }
+  };
+
+  reversed = (event: React.MouseEvent<HTMLElement>) => {
+    const target = event.currentTarget as Element;
+    const parentEl = target.parentElement as Element;
+
+    const alphButton = parentEl.children[0] as Element;
+    const lenghtButton = parentEl.children[1] as Element;
+
+    if (target.classList.contains('is-light')) {
+      this.setState({
+        showResetButton: true,
+      });
+    } else if (alphButton.classList.contains('is-light')
+        && lenghtButton.classList.contains('is-light')) {
+      this.setState({
+        showResetButton: false,
+      });
+    }
+
+    this.setState(prevState => ({
+      isReversed: !prevState.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+      showResetButton: false,
+    });
+  };
+
+  render() {
+    const { sortType, isReversed, showResetButton } = this.state;
+
+    const reorderedGoods = getReorderedGoods(
+      goodsFromServer,
+      { sortType, isReversed },
+    );
+
+    // const removeBut = document.querySelector(".is-danger");
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            onClick={this.alphSort}
+            type="button"
+            className={classNames(
+              'button', 'is-info',
+              {
+                'is-light': sortType !== 'alphabet',
+              },
+            )}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            onClick={this.lengthSort}
+            type="button"
+            className={classNames(
+              'button', 'is-success',
+              {
+                'is-light': sortType !== 'length',
+              },
+            )}
+          >
+            Sort by length
+          </button>
+
+          <button
+            onClick={this.reversed}
+            type="button"
+            className={classNames(
+              'button', 'is-warning',
+              {
+                'is-light': isReversed !== true,
+              },
+            )}
+          >
+            Reverse
+          </button>
+
+          {showResetButton
```

Comment:
```suggestion
          {(sortType !== SortType.NONE || isReversed)
```

Reply:
Thank you, of course now it seems easier and works the same)